src/firebolt/common/token_storage.py

```diff
@@ -0,0 +1,114 @@
+import base64
+import json
+import os
+from json import JSONDecodeError
+from typing import Optional
+
+from appdirs import user_data_dir
+from cryptography.fernet import Fernet, InvalidToken
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.kdf.pbkdf2 import PBKDF2HMAC
+
+
+class TokenSecureStorage:
+    def __init__(self, username: str, password: str):
+        """
+        Class for permanent storage of token in the filesystem in encrypted way
+
+        :param username: username used for toke encryption
+        :param password: password used for toke encryption
+        """
+        self._data_dir = user_data_dir(appname="firebolt")
+        os.makedirs(self._data_dir, exist_ok=True)
+
+        self._token_file = os.path.join(self._data_dir, "token.json")
```

Comment:
Seems like this implementation only supports caching token for a single set of credentials.
I would suggest changing `token.json` to something like `hash(username + password).json`, which would resolve this issue